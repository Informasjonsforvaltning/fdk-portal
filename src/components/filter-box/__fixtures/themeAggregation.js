export const themeAggregation = {
  buckets: [
    { key: 'GOVE', count: 104 },
    { key: 'EDUC', count: 11 },
    { key: 'HEAL', count: 8 },
    { key: 'Ukjent', count: 7 },
    { key: 'AGRI', count: 1 },
    { key: 'TEST', count: 1 }
  ]
};
