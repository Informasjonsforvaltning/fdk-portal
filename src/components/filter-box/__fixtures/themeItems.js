export const themeItems = {
  AGRI: {
    id: 'http://publications.europa.eu/resource/authority/data-theme/AGRI',
    code: 'AGRI',
    startUse: '2015-10-01',
    title: {
      pt: 'Agricultura, pesca, silvicultura e alimentação',
      bg: 'Селско стопанство, рибарство, горско стопанство и храни'
    },
    conceptSchema: {
      id: 'http://publications.europa.eu/resource/authority/data-theme',
      title: { en: 'Dataset types Named Authority List' }
    }
  }
};
