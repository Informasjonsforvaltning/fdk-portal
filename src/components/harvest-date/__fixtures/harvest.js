export const harvest = {
  firstHarvested: '2018-06-25T19":51:05+0200',
  lastHarvested: '2018-07-09T14:40:37+0200',
  lastChanged: '2018-07-09T14:40:37+0200',
  changed: [
    '2018-06-25T19:51:05+0200',
    '2018-06-25T19:51:12+0200',
    '2018-06-26T09:16:46+0200',
    '2018-06-28T09:22:41+0200',
    '2018-06-28T09:52:24+0200',
    '2018-06-29T15:25:53+0200',
    '2018-07-03T10:59:38+0200',
    '2018-07-09T14:40:37+0200'
  ]
};
