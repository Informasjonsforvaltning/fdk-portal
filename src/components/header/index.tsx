import React from 'react';
import type { FC } from 'react';
import { AppNavBar } from '../../app/app-nav-bar/app-nav-bar';
import localization from '../../lib/localization';

interface Props {
  onChangeLanguage: (language: string) => void;
}

const Header: FC<Props> = ({ onChangeLanguage }) => (
  <div>
    <div>
      <a
        id='focus-element'
        className='uu-invisible'
        href={`${location.pathname}#content`}
        aria-hidden='true'
      >
        {localization.goToMainContent}
      </a>
    </div>
    <AppNavBar onChangeLanguage={onChangeLanguage} />
  </div>
);

export default Header;
