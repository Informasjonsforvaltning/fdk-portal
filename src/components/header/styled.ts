import styled from 'styled-components';
import { Colour, theme as themeFdk } from '@fellesdatakatalog/theme';

const Header = styled.div`
  border-bottom: 1px solid ${themeFdk.colour(Colour.NEUTRAL, 'N30')};
`;

export default {
  Header
};
