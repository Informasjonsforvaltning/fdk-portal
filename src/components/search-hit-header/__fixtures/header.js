export const header = {
  title: 'Test tittel',
  publisherLabel: 'Eies av',
  publisher: {
    id: '974760673'
  }
};
