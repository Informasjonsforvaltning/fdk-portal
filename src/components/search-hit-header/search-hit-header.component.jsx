import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { Link } from 'react-router-dom';
import cx from 'classnames';

import { LabelStatus } from '../label-status/label-status.component';
import { PublisherLabel } from '../publisher-label/publisher-label.component';
import { getPublisherByOrgNr } from '../../redux/modules/publishers';
import { getLosStructure } from '../../redux/modules/referenceData';
import { getTranslateText } from '../../lib/translateText';
import localization from '../../lib/localization';
import './search-hit-header.scss';
import { LabelNational } from '../label-national/label-national.component';
import { LinkExternal } from '../link-external/link-external.component';
import { getConfig } from '../../config';

const renderPublisher = (
  publisherLabel,
  publisher,
  publisherItems,
  catalog
) => {
  if (!publisher) {
    return null;
  }
  const publisherItem =
    getPublisherByOrgNr(publisherItems, _.get(publisher, 'id')) || publisher;
  return (
    <PublisherLabel
      label={publisherLabel}
      publisherItem={publisherItem}
      catalog={catalog}
    />
  );
};

const renderThemes = (themes, losItems, darkThemeBackground) => {
  const themeClass = cx('align-self-center mr-2 mb-2', {
    'fdk-label': !darkThemeBackground,
    'fdk-label-details': darkThemeBackground
  });

  return themes
    .map((singleTheme, index) => {
      const losItem = _.find(losItems, { uri: singleTheme.id });
      const title = getTranslateText(
        _.get(losItem, 'prefLabel') || singleTheme.title
      );
      return (
        title && (
          <div
            key={`dataset-description-theme-${index}`}
            className={themeClass}
          >
            <span className="uu-invisible" aria-hidden="false">
              Datasettets tema.
            </span>
            {title}
          </div>
        )
      );
    })
    .filter(Boolean);
};

const renderTitle = (Tag, title, titleLink, externalLink) => {
  const titleTag = (Tag, title) => (
    <Tag className="mr-3" name={title}>
      {title}
    </Tag>
  );
  if (titleLink) {
    if (externalLink) {
      return (
        <LinkExternal
          uri={getConfig().searchHost.host.concat(titleLink)}
          prefLabel={title}
        />
      );
    }
    return (
      <Link
        className="search-hit__title-link"
        title={`${localization.apiLabel}: ${title}`}
        to={titleLink}
      >
        {titleTag(Tag, title)}
      </Link>
    );
  }
  return titleTag(Tag, title);
};

export const SearchHitHeader = props => {
  const {
    tag: Tag,
    title,
    titleLink,
    catalog,
    publisherLabel,
    publisher,
    publisherTag,
    publisherItems,
    theme: themes = [],
    nationalComponent,
    statusCode,
    referenceData,
    darkThemeBackground,
    externalLink
  } = props;

  const losItems = getLosStructure(referenceData);
  return (
    <>
      {title && (
        <div className="mb-2 d-flex flex-wrap align-items-center">
          {renderTitle(Tag, title, titleLink, externalLink)}
          {statusCode && (
            <LabelStatus
              statusCode={statusCode}
              referenceData={referenceData}
            />
          )}
        </div>
      )}

      <div className="mb-4 d-flex flex-wrap align-items-baseline">
        {publisherItems &&
          renderPublisher(publisherLabel, publisher, publisherItems, catalog)}

        {!publisherItems && publisher && (
          <PublisherLabel
            tag={publisherTag}
            label={publisherLabel}
            publisherItem={publisher}
            catalog={catalog}
          />
        )}
      </div>

      {(nationalComponent || !!themes.length) && (
        <div className="mb-4 d-flex flex-wrap align-items-baseline align-items-center">
          {nationalComponent && <LabelNational />}
          {renderThemes(themes, losItems, darkThemeBackground)}
        </div>
      )}
    </>
  );
};

SearchHitHeader.defaultProps = {
  tag: 'h1',
  title: null,
  titleLink: null,
  publisherLabel: null,
  publisher: null,
  publisherTag: 'span',
  publisherItems: null,
  theme: [],
  nationalComponent: false,
  statusCode: null,
  referenceData: null,
  darkThemeBackground: false,
  externalLink: false
};

SearchHitHeader.propTypes = {
  tag: PropTypes.string,
  title: PropTypes.string,
  titleLink: PropTypes.string,
  publisherLabel: PropTypes.string,
  publisher: PropTypes.object,
  publisherTag: PropTypes.string,
  publisherItems: PropTypes.object,
  theme: PropTypes.array,
  nationalComponent: PropTypes.bool,
  statusCode: PropTypes.string,
  referenceData: PropTypes.object,
  darkThemeBackground: PropTypes.bool,
  externalLink: PropTypes.bool
};
