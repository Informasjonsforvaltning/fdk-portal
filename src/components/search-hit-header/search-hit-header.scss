@import 'designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.search-hit__title-link {
  text-decoration: none;
  @include themify($themes) {
    color: themed('color-text-link');
  }
}

.fdk-label {
  @include themify($themes) {
    background-color: themed('color-neutral-lightest');
    color: themed('color-neutral-dark');
  }
  font-size: 1.4rem !important;
  border-radius: .9em !important;
  padding: .0em .6em .0em .6em !important;
  display: inline-block !important;
  line-height: 1.5em !important;
}

.fdk-national-component {
  @include themify($themes) {
    background-color: themed('color-neutral-darkest');
    color: themed('color-neutral-lightest');
  }
  font-size: 1.4rem !important;
  border-radius: .9em !important;
  padding: .0em .6em .0em .6em !important;
  display: inline-block !important;
  line-height: 1.5em !important;
}

.fdk-label-details {
  @include themify($themes) {
    background-color: themed('color-neutral-light');
    color: themed('color-neutral-darkest');
  }
  font-size: 1.4rem !important;
  border-radius: .9em !important;
  padding: .0em .6em .0em .6em !important;
  display: inline-block !important;
  line-height: 1.5em !important;
}

.fdk-national-component-icon {
  background-image: url("/img/icon-authoritative-sm.svg");
  width: 1em; 
  height: 1em;
  display: inline-block;
}
