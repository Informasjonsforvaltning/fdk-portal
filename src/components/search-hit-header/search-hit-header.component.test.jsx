import React from 'react';
import { shallow } from 'enzyme';
import { SearchHitHeader } from './search-hit-header.component';
import { header } from './__fixtures/header';
import publishers from '../../../test/fixtures/publishers';

test('should render SearchHitHeader with no props', () => {
  const wrapper = shallow(<SearchHitHeader />);
  expect(wrapper).toMatchSnapshot();
});

test('should render SearchHitHeader with harvest dates', () => {
  const wrapper = shallow(
    <SearchHitHeader {...header} publisherItems={publishers} />
  );
  expect(wrapper).toMatchSnapshot();
});
