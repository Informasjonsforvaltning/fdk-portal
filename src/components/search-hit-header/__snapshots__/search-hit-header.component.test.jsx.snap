// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchHitHeader with harvest dates 1`] = `
<Fragment>
  <div
    className="mb-2 d-flex flex-wrap align-items-center"
  >
    <h1
      className="mr-3"
      name="Test tittel"
    >
      Test tittel
    </h1>
  </div>
  <div
    className="mb-4 d-flex flex-wrap align-items-baseline"
  >
    <PublisherLabel
      label="Eies av"
      publisherItem={
        Object {
          "id": "974760673",
          "naeringskode": Object {
            "code": "84.110",
            "prefLabel": Object {
              "no": "Generell offentlig administrasjon",
            },
            "uri": "http://www.ssb.no/nace/sn2007/84.110",
          },
          "name": "Brønnøysundregistrene",
          "orgPath": "/STAT/912660680/974760673",
          "organisasjonsform": "ORGL",
          "overordnetEnhet": "912660680",
          "sektorkode": Object {
            "code": "6100",
            "prefLabel": Object {
              "no": "Statsforvaltningen",
            },
            "uri": "http://www.brreg.no/sektorkode/6100",
          },
          "uri": "http://data.brreg.no/enhetsregisteret/enhet/974760673",
          "valid": true,
        }
      }
      tag="span"
    />
  </div>
</Fragment>
`;

exports[`should render SearchHitHeader with no props 1`] = `
<Fragment>
  <div
    className="mb-4 d-flex flex-wrap align-items-baseline"
  />
</Fragment>
`;
